File: user-service/src/models/http-response.ts
import { StatusCodes } from "http-status-codes";

export interface IHeaders {
    [key: string]: string;
}

export interface IHttpResponse<T> {
    statusCode: number,
    headers?: IHeaders,
    body: T
}

export interface IError {
    message: string;
}

export const jsonResponse = <T>(body: T, statusCode: StatusCodes = StatusCodes.OK): IHttpResponse<T> => {
    return {
        statusCode, body,
        headers: {
            'Content-Type': 'application/json'
        }
    }
}

File: auth-service/http-response.go
package main

import "fmt"

// IHttpResponse models a http response message payload
type IHttpResponse struct {
	statusCode int
	headers    map[string]string
	body       interface{}
}

// IError models a generic error message
type IError struct {
	Message string `json:"message"`
}

// Need to implement the Error() method for IError
// to be considered to implement the error interface
func (e IError) Error() string {
	return fmt.Sprintf("%s", e.Message)
}
